### PR TITLE
chore(deps): update deps roll-up Feb-04-2017

### PR DESCRIPTION
**Pull request by gihrig**
-- Test script --
yarn && npm start
Test Dev, Dev tunnel, Features, lang = de, NotFound, Production, off-line
npm run clean
npm run generate component TestComp /w defaults
npm run generate container TestPage /w defaults
npm run generate route to TestPage - /test
use TestComp on TestPage
npm bypass all tests in TestComp and TestPage (set true = true)
npm start > localhost:3000/test
npm test (expect test failures due to incomplete test coverage)
npm run build
npm run start:prod (Run "Test Dev..." as above)
git add .
git commit -m "testing..."

Node 7.5.0
npm 4.1.2
yarn 0.20.0
Mac OS 10.11.6
Chrome 56.0.2924.87 (64-bit)

chore(deps): update deps roll-up Feb-04-2017

- **Patch Updates:**
 express                                         4.14.0  →  4.14.1 
 fontfaceobserver                                 2.0.7  →   2.0.8 
 redux-immutable                                 3.0.10  →  3.0.11 
 babel-plugin-react-intl                          2.3.0  →   2.3.1 
 babel-plugin-transform-react-remove-prop-types  0.2.11  →  0.2.12 
 enzyme                                           2.7.0  →   2.7.1 
 eslint                                          3.13.0  →  3.13.1 
 lint-staged                                      3.2.7  →   3.2.9 
 ngrok                                            2.2.5  →   2.2.6 
 node-plop                                        0.5.4  →   0.5.5 
 plop                                             1.7.3  →   1.7.4 
 webpack                                          2.2.0  →   2.2.1 

- **Minor updates:**
 history                  3.0.0  →   3.2.1 
 styled-components        1.3.1  →   1.4.3 
 file-loader              0.9.0  →  0.10.0 
 html-webpack-plugin     2.26.0  →  2.28.0 
 lint-staged              3.2.9  →   3.3.0 
 offline-plugin           4.5.5  →   4.6.1 
 webpack-dev-middleware   1.9.0  →  1.10.0 
 webpack-hot-middleware  2.15.0  →  2.16.1 

 - Incompatible per `npm info "eslint-config-airbnb@latest" peerDependencies` 
   adjust eslint-config-airbnb compatible dependency versions. 
   see: https://www.npmjs.com/package/eslint-config-airbnb

   ***eslint                  3.13.1  →  3.15.0 <--- rolled back***

- **Major updates:**
 none

 - Error observed in Travis CI build
```bash
   ERROR in ./~/react-router-scroll/lib/StateStorage.js
   Module not found: Error: Can't resolve 'history/lib/DOMStateStorage' in '
   /home/travis/build/gihrig/react-boilerplate/node_modules/react-router-scroll/lib'
    @ ./~/react-router-scroll/lib/StateStorage.js 5:23-61
    @ ./~/react-router-scroll/lib/ScrollBehaviorContext.js
    @ ./~/react-router-scroll/lib/useScroll.js
    @ ./~/react-router-scroll/lib/index.js
    @ ./app/app.js
    @ multi ./app/app.js
```

   react-boilerplate@3.4.0 /Users/glen/Documents/JavaScript/Research/react-bp
   └── UNMET PEER DEPENDENCY history@4.5.1

   npm ERR! peer dep missing: history@^2.0.0 || ^3.0.0, required by react-router-scroll@0.4.1

   ***history               3.2.1  →   4.5.1 <--- rolled back***

 Addressed here https://github.com/imagemin/imagemin-pngquant/issues/32 ***Needs Testing***
 - Error: (some OS X installations only, not version specific)

   Addressed here https://github.com/imagemin/imagemin-pngquant/issues/32, but throws warnings (below)

   In ./app/components/Header/banner.jpg
   Module build failed: Error: dyld: Library not loaded: /usr/local/opt/libpng/lib/libpng16.16.dylib
     Referenced from: ~/react-boilerplate/node_modules/mozjpeg/vendor/cjpeg
     Reason: image not found

   ***image-webpack-loader              2.0.0  →    3.0.0 <--- rolled back***
   see: https://github.com/mxstbr/react-boilerplate/pull/1322#issuecomment-266264527 and #1335  

 - Warning: Observed on `npm start`
  WARNING in ./app/components/Header/banner.jpg
   DEPRECATED. Configure gifsicle's interlaced option in it's own options. (gifsicle.interlaced)
    @ ./app/components/Header/index.js 47:0-34
    @ ./app/containers/App/index.js
    @ ./app/app.js
    @ multi eventsource-polyfill webpack-hot-middleware/client?reload=true ./app/app.js
  WARNING in ./app/components/Header/banner.jpg
   DEPRECATED. Configure optipng's optimizationLevel option in it's own options. (optipng.optimizationLevel)
   @ ./app/components/Header/index.js 47:0-34
   @ ./app/containers/App/index.js
   @ ./app/app.js
   @ multi eventsource-polyfill webpack-hot-middleware/client?reload=true ./app/app.js

   ***image-webpack-loader  2.0.0  →   3.2.0 <--- Rolled back***

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -261,27 +261,27 @@
     "coveralls": "2.11.15",
     "css-loader": "0.26.1",
     "enzyme": "2.7.1",
-    "eslint": "3.14.1",
-    "eslint-config-airbnb": "14.0.0",
+    "eslint": "^3.13.0",
```
Comment (samit4me) [inline package.json]: Did you mean to add the caret's?

Comment (gihrig) [inline package.json]: No that's an error. 

I used the script on [this page](https://www.npmjs.com/package/eslint-config-airbnb) to ensure eslint-config-airbnb and friends played nicely. Didn't see the carets. Will fix shortly...

 Thanks for reviewing 😄

**Comment by gihrig** [comment]
>What is our process now given no more LGTM? Can one dev approve and merge (this looks possible), or are we still wanting two approvals?

Technically, I think we are waiting for two, but that doesn't seem to happen lately 😢 

So I have waited for at least a 48hrs, and if there is no further input from the community and I have a need for the change, I'll go ahead and commit the PR, if that's possible.

I'm going on the idea that the community is neutral on the issue, so if it matters to me, at least there are no objections. Not the greatest community involvement, but for relatively simple changes I think it's OK.

Re: this PR, it's already over a week old and there is a problem with dependency updates in that the older they get the more complex testing and correction becomes, just because there are more changes. So I I'm not doing a new deps update this week because this was not approved sooner, knowing full well that it may make the next one more difficult.

I think the solution lies in fully automated end-to-end testing, but I don't have the time to figure that out just now (although I would love to have it in my projects).

**Comment by samit4me** [comment]
Thanks for the info. I know the pain regarding updating dependancies, I've got that job at work and you're right, do it regularly and it's manageable, but leave it for a few weeks and it can be difficult.

I too like the idea of two people approving will signal its okay to merge, but given that the workflow only has a single approve it's kinda difficult.

**Comment by gihrig** [comment]
Thanks for checking, thanks for merging. ✨ 

I just signed up as an early adopter at [Cypress.io](https://www.cypress.io/). The early stage promo looks pretty good for easier end-to-end testing. (fingers crossed) 😉

**Comment by samit4me** [comment]
That's the first I've heard of Cypress, but from first impressions, it looks good.

> No. Cypress replaces Selenium completely

^ I was sold after I read that :tada:

**Pull request merged**